packages/eas-cli/src/build/utils/repository.ts

```diff
@@ -83,9 +83,54 @@ export async function commitPromptAsync(
   });
 }
 
-export async function makeProjectTarballAsync(
-  vcsClient: Client
-): Promise<{ path: string; size: number }> {
+export type LocalFile = {
+  path: string;
+  size: number;
+};
+
+export async function makeProjectMetadataFileAsync(archivePath: string): Promise<LocalFile> {
+  const spinner = ora('Creating project metadata file');
+  const timerLabel = 'makeProjectMetadataFileAsync';
+  const timer = setTimeout(
+    () => {
+      spinner.start();
+    },
+    Log.isDebug ? 1 : 1000
+  );
+  startTimer(timerLabel);
+
+  const metadataLocation = path.join(getTmpDirectory(), `${uuidv4()}-eas-build-metadata.json`);
```

Comment:
Is the filename preserved — do GCS files have the same name? Maybe it would be better to call them `eas-build-metadata-${uuidv4()}.json` if that's the case?